dt-posts/dt-posts.php

```diff
@@ -1423,6 +1423,52 @@ public static function get_post_tiles( $post_type, $return_cache = true ){
         return $tile_options[$post_type];
     }
 
+    /**
+     * Request record access
+     *
+     * @param string $post_type
+     * @param int $post_id
+     * @param int $user_id -- Requesting user
+     *
+     * @return false|int|WP_Error
+     */
+    public static function request_record_access( string $post_type, int $post_id, int $user_id ) {
+
+        // Sanity checks
+        if ( ! in_array( $post_type, self::get_post_types() ) ) {
```

Comment:
This might be a good upgrade, checks the post type and that the user has access
```
if ( !self::can_access( $post_type ) ) {
       return new WP_Error( __FUNCTION__, sprintf( "You do not have access to these %s", $post_type ), [ 'status' => 403 ] );
}
```